en/08_Changelogs/6.0.0.md

```diff
@@ -29,6 +30,242 @@ title: 6.0.0 (unreleased)
+
+#### Changes to `/dev/*` actions {#cli-dev}
+
+With the changes to `sake` come changes to the way `dev/*` actions are handled. Most of these are now subclasses of the new [`DevCommand`](api:SilverStripe\Dev\Command\DevCommand) class which is itself a subclass of `PolyCommand`.
+
+If you have custom actions registered under `DevelopmentAdmin.registered_controllers` you'll need to update the YAML configuration for these. If you want them to be accessible via CLI, you'll also have to update the PHP code.
+
+With the below example, there are two custom actions displayed in the list at `/dev`:
+
+- `/dev/my-action`: intended for use in the browser only, but you'd have to add custom logic in `init()` to disallow its use in CLI until now
+- `/dev/my-other-action`: intended for use both in CLI and in the browser.
+
+For actions that should only be accessible in the browser, you only need to change how these are registered. Move them from `DevelopmentAdmin.registered_controllers` to the new [`DevelopmentAdmin.controllers`](api:SilverStripe\Dev\DevelopmentAdmin->controllers) configuration property.
+
+Controllers added to `DevelopmentAdmin.registered_controllers` can only be accessed via HTTP requests, so you can remove any logic around CLI usage.
+
+For actions that should be accessible in the browser *and* via CLI, you will need to change these from being a `Controller` to subclassing `DevCommand`. These get registered to the new [`DevelopmentAdmin.commands`](api:SilverStripe\Dev\DevelopmentAdmin->commands) configuration property
+
+```diff
+ SilverStripe\Dev\DevelopmentAdmin:
+-  registered_controllers:
+-    my-action:
+-      controller: 'App\Dev\MyActionController'
+-      links:
+-        my-action: 'Perform my custom action in dev/my-action (do not run in CLI)'
+-    my-other-action:
+-      controller: 'App\Dev\MyOtherActionController'
+-      links:
+-        my-other-action: 'Perform my custom action in dev/my-other-action'
++  controllers:
++    my-action:
++      class: 'App\Dev\MyActionController'
++      description: 'Perform my custom action in dev/my-action'
++  commands:
++    my-other-action: 'App\Dev\MyOtherActionCommand'
+```
+
+```diff
+ namespace App\Dev;
+
+-use SilverStripe\Control\Controller;
+-use SilverStripe\Control\Director;
+-use SilverStripe\Control\HTTPRequest;
++use SilverStripe\Dev\Command\DevCommand;
+-use SilverStripe\Dev\DevelopmentAdmin;
++use SilverStripe\PolyExecution\PolyOutput;
+-use SilverStripe\Security\Permission;
+ use SilverStripe\Security\PermissionProvider;
+-use SilverStripe\Security\Security;
++use Symfony\Component\Console\Command\Command;
++use Symfony\Component\Console\Input\InputInterface;
++use Symfony\Component\Console\Input\InputOption;
+
+-class MyOtherActionController extends Controller implements PermissionProvider
++class MyOtherActionController extends DevCommand implements PermissionProvider
```

Comment:
```suggestion
+class MyHttpAndCliActionController extends DevCommand implements PermissionProvider
```

Reply:
Done - also updated `MyActionController` to `MyHttpOnlyActionController`